### PR TITLE
Update Chromedriver to v114

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

This avoids version mismatch errors when running e2e tests with Chrome locally.

Note that in order to make this change, we have to instruct LavaMoat to ignore two more globals which are used internally by Chromedriver:

- `Proxy`
- `ret_nodes`

The second global definitely seems like something specific to Chromedriver, but the first one — Proxy — is part of EcmaScript. Always ignoring this seems to go against the spirit of LavaMoat. To address this, then, this commit not only upgrades Chromedriver but also adds a guard around these two exceptions — along with other Chromedriver-specific LavaMoat exceptions we've added in the past — so that they only apply for test builds. This ensure that in the production version of MetaMask, access to Proxy is properly sandboxed.

Inspired by https://github.com/MetaMask/metamask-extension/pull/19180.

## Manual Testing Steps

You should be able to run all of the Chrome e2e tests locally. In addition, CI should pass.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by DDDDDanica — DISMISSED** [review]

File: development/build/index.js
Comment (Gudahtt) [inline development/build/index.js:113]: There are two entry tasks for test builds: "test" and "testDev".

We also have constants for each of them. I have another PR open now where I've replaced the other entry task references in this file with constants: https://github.com/MetaMask/metamask-extension/pull/19506

**Comment by Gudahtt** [comment]
Creating a separate list of scuttling exceptions for test builds is a great idea. However, the current list of supposedly Chromedriver-related scuttling exceptions is a bit suspect. If there was an entry there relied upon in a production build, it wouldn't be caught by CI. As such we should probably do some manual testing on the production build before reducing that list.

File: development/build/index.js
Comment (mcmire) [reply development/build/index.js:113]: Ah, okay. I was going off the `shouldScuttle` array above. Your changes make sense. I'll wait until that PR is merged and then rebase this.

**Comment by mcmire** [comment]
@Gudahtt Right, fair enough. I can do that.

**Comment by mcmire** [comment]
@Gudahtt Ah sorry I misread your comment. It sounds like reducing the list can be something we do later. For now I've only added the `if` statement around `Proxy` and `ret_nodes`. Hopefully that addresses your concerns.

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by brad-decker — DISMISSED** [review]

File: development/build/index.js
Comment (mcmire) [inline development/build/index.js:129]: It appears that `Proxy` may have slipped into https://github.com/MetaMask/metamask-extension/pull/19293 by accident, so I moved it back out.

**Comment by mcmire** [comment]
^ Fixed conflicts and re-pushed.

**Review by Gudahtt — DISMISSED** [review]

**Comment by mcmire** [comment]
Sorry one more approval if you please ^

**Review by Gudahtt — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Review by PeterYinusa — APPROVED** [review]

**Pull request merged**
